core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/deletevolumes/DeleteVolumesService.java

```diff
@@ -0,0 +1,118 @@
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudPlatformVariant;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.Platform;
+import com.sequenceiq.cloudbreak.cloud.model.Variant;
+import com.sequenceiq.cloudbreak.cluster.api.ClusterApi;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.CommonUtilService;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.domain.Resource;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.reactor.api.event.resource.DeleteVolumesRequest;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.resource.ResourceService;
+import com.sequenceiq.cloudbreak.template.model.ServiceComponent;
+import com.sequenceiq.common.api.type.ResourceType;
+
+@Service
+public class DeleteVolumesService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeleteVolumesService.class);
+
+    @Inject
+    private CloudbreakFlowMessageService flowMessageService;
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private ResourceService resourceService;
+
+    @Inject
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Inject
+    private CommonUtilService commonUtilService;
+
+    public void deleteVolumes(StackDeleteVolumesRequest payload) {
+        flowMessageService.fireEventAndLog(payload.getStackId(),
+                Status.UPDATE_IN_PROGRESS.name(),
+                CLUSTER_DELETINGVOLUMES,
+                payload.getGroup(),
+                payload.getStackId().toString());
+    }
+
+    public void detachResources(List<CloudResource> cloudResourcesToBeDetached, String cloudPlatform, AuthenticatedContext ac) throws Exception {
+        LOGGER.debug("Detaching volumes {}", cloudResourcesToBeDetached);
+        CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(Platform.platform(cloudPlatform), Variant.variant(cloudPlatform));
+        CloudConnector cloudConnector = cloudPlatformConnectors.get(cloudPlatformVariant);
+        cloudConnector.volumeConnector().detachVolumes(ac, cloudResourcesToBeDetached);
+    }
+
+    public void deleteResources(List<CloudResource> cloudResourcesToBeDeleted, String cloudPlatform, AuthenticatedContext ac) throws Exception {
+        LOGGER.debug("Deleting volumes {}", cloudResourcesToBeDeleted);
+        CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(Platform.platform(cloudPlatform), Variant.variant(cloudPlatform));
+        CloudConnector cloudConnector = cloudPlatformConnectors.get(cloudPlatformVariant);
+        cloudConnector.volumeConnector().deleteVolumes(ac, cloudResourcesToBeDeleted);
+    }
+
+    public void updateCbdbResources(StackDto stackDto, DeleteVolumesRequest payload) throws Exception {
```

Comment:
I would rather name it as `deleteVolumeResources`, also would update the same in the following log message.

Reply:
done